Assets/Scripts/PlayerCharacter/PlayerCharacterController.cs

```diff
@@ -27,8 +42,33 @@ private void FixedUpdate()
             float horiz = Input.GetAxisRaw("Horizontal");
             float vert = Input.GetAxisRaw("Vertical");
 
-            Vector2 direction= new Vector2(horiz, vert).normalized;
+            Vector2 direction = new Vector2(horiz, vert).normalized;
             rigidBody.AddForce(direction * acceleration);
+
+            //controller.DisableInputForSeconds(paralyzeDuration); // disable input for stalling effect
+
+            PowerUpTimer(); // Consuming Caffeine adds time duration and runs its timer
+        }
+                
+        public void PowerUpDuration(int amount, float duration)
+        {
+            this.powerUpValue += amount;
+            this.powerUpTimer += duration;
+        }
+
+        private void PowerUpTimer()
```

Comment:
This is a separate thing to controlling the movement of player character, which is what `PlayerCharacterController` is trying to do. So it would be better if this gets its own component, something like `CaffeineEffectTracker` or so.